networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/PeerSubnetSubscriptions.java

```diff
@@ -55,42 +56,69 @@ public static PeerSubnetSubscriptions create(
-      final int targetSubnetSubscriberCount) {
+      final int targetSubnetSubscriberCount,
+      final SettableLabelledGauge subnetPeerCountGauge) {
     final Map<String, Collection<NodeId>> subscribersByTopic = network.getSubscribersByTopic();
 
-    return builder(currentSchemaDefinitions)
-        .targetSubnetSubscriberCount(targetSubnetSubscriberCount)
-        .attestationSubnetSubscriptions(
-            b ->
-                // Track all attestation subnets
-                streamAllAttestationSubnetIds(currentSchemaDefinitions)
-                    .forEach(
-                        attestationSubnet -> {
-                          b.addRelevantSubnet(attestationSubnet);
-                          subscribersByTopic
-                              .getOrDefault(
-                                  attestationTopicProvider.getTopicForSubnet(attestationSubnet),
-                                  Collections.emptySet())
-                              .forEach(
-                                  subscriber -> b.addSubscriber(attestationSubnet, subscriber));
-                        }))
-        .syncCommitteeSubnetSubscriptions(
-            b ->
-                // Only track sync committee subnets that we're subscribed to
-                syncCommitteeSubnetService
-                    .getSubnets()
-                    .forEach(
-                        syncCommitteeSubnet -> {
-                          b.addRelevantSubnet(syncCommitteeSubnet);
-                          subscribersByTopic
-                              .getOrDefault(
-                                  syncCommitteeSubnetTopicProvider.getTopicForSubnet(
-                                      syncCommitteeSubnet),
-                                  Collections.emptySet())
-                              .forEach(
-                                  subscriber -> b.addSubscriber(syncCommitteeSubnet, subscriber));
-                        }))
-        .build();
+    final PeerSubnetSubscriptions subscriptions =
+        builder(currentSchemaDefinitions)
+            .targetSubnetSubscriberCount(targetSubnetSubscriberCount)
+            .attestationSubnetSubscriptions(
+                b ->
+                    // Track all attestation subnets
+                    streamAllAttestationSubnetIds(currentSchemaDefinitions)
+                        .forEach(
+                            attestationSubnet -> {
+                              b.addRelevantSubnet(attestationSubnet);
+                              subscribersByTopic
+                                  .getOrDefault(
+                                      attestationTopicProvider.getTopicForSubnet(attestationSubnet),
+                                      Collections.emptySet())
+                                  .forEach(
+                                      subscriber -> b.addSubscriber(attestationSubnet, subscriber));
+                            }))
```

Comment:
and i guess further, did we want to set to something to know its not subscribed... i guess that might be a bit random...

Reply:
It's an interesting point actually. We're iterating all the attestation subnets all the time so that's not an issue, but sync committee subnets we only iterate when we actually need them, so potentially we'd fail to reset though.  Will take a look and see what we can do.

Reply:
Fixed so now we iterate through all subnets for both attestations and sync committees when updating the metrics.

Reply:
sounds like a good plan!